mne_connectivity/spectral/epochs.py

```diff
@@ -189,6 +196,46 @@ def _prepare_connectivity(epoch_block, times_in, tmin, tmax,
             n_signals, indices_use, warn_times)
 
 
+def _check_rank_input(rank, data, indices):
+    """Check the rank argument is appropriate and compute rank if missing."""
+    sv_tol = 1e-10  # tolerance for non-zero singular val (rel. to largest)
```

Comment:
I would go with something more conservative like `1e-6` if possible (data are often stored in single precision and 1e-10 will always look "full rank"). And in any case if you work with M/EEG data for example you'll need to scale with an ad-hoc or empirical noise cov whitener to get things on roughly the same scale

Reply:
Completely agree. I came across a few cases earlier this week using the code where `1e-10` was too lenient which to led to some `LinAlgError`s being caught later in the Granger causality computations. Have updated as such:
https://github.com/mne-tools/mne-connectivity/blob/883e816e097c1894008ac088733910b8aca87d29/mne_connectivity/spectral/epochs.py#L199-L201